src/client/Microsoft.Identity.Client/AuthScheme/PoP/PoPAuthenticationScheme.cs

```diff
@@ -67,25 +67,26 @@ public IReadOnlyDictionary<string, string> GetTokenRequestParams()
             };
         }
 
-        public string FormatAccessToken(MsalAccessTokenCacheItem msalAccessTokenCacheItem)
+        public void FormatResult(AuthenticationResult authenticationResult)
```

Comment:
I think this will work, now that the additionalCacheParameters are stored in the `AuthenticaitonResult.AdditionalResponseParameters`, the ds_req_cnf and the nonce will be available to use.